lightning/src/routing/network_graph.rs

```diff
@@ -660,6 +659,23 @@ pub struct ChannelInfo {
 	announcement_received_time: u64,
 }
 
+impl ChannelInfo {
+	/// Returns a [`DirectedChannelInfo`] for the channel directed to the given `target`, or `None`
+	/// if `target` is not one of the channel's counterparties.
+	pub fn as_directed_to(&self, target: &NodeId) -> Option<DirectedChannelInfo> {
+		let (direction, source, target) = {
+			if target == &self.node_one {
```

Comment:
Could we also skip entries where `self.two_to_one` is `None` (similar below), then make `direction` a non-`Option` field of `DirectedChannelInfo`? Would omit the need for the `unwrap`s later and seems we wouldn't use it if it's `None` anyway

Reply:
Hmm... so this is also used later by `ProbabilisticScorer` in event handling, where the `NetworkGraph` may have changed in some manner. Perhaps the `ChannelUpdate` has gone stale and was removed. We'd still want to use the capacity from `ChannelInfo` there, for instance.

Reply:
Testing this would be a bit difficult and wouldn't catch new uses. Ended having the compiler enforce this by defining a wrapper that can only be created if `direction` is `Some`.